check/src/unify_type.rs

```diff
@@ -68,6 +74,21 @@ impl<I> fmt::Display for TypeError<I>
                         outside its scope",
                        id)
             }
+            TypeError::MissingFields(ref typ, ref fields) => {
+                try!(write!(f, "The type `{}` lacks the following fields: ", typ));
+                for (i, field) in fields.iter().enumerate() {
+                    let sep = if i == 0 {
+                        ""
+                    } else if i == fields.len() - 1 {
+                        " and "
+                    } else {
+                        ", "
+                    };
```

Comment:
I can never decide on the best way to write these lists. Might be useful to have some abstraction over 'listing things in a human readable way'.